src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -412,17 +388,19 @@ def start_servers(self, server_groups=None):
             # Optionally start servers on a different subset of hosts with a
             # different server group
             for group, hosts in server_groups.items():
+                self.server_manager = ServerManager(
+                    self.basepath, os.path.join(self.ompi_prefix, "bin"))
                 self.log.info(
                     "Starting servers: group=%s, hosts=%s", group, hosts)
-                hostfile = write_host_file.write_host_file(hosts, self.workdir)
-                server_utils.run_server(
-                    hostfile, group, self.basepath,
-                    log_filename=self.server_log)
-
+                self.server_manager.hosts = (
+                    hosts, self.workdir, self.hostfile_servers_slots)
+                self.server_manager.start(log_file=self.server_log)
         else:
-            server_utils.run_server(
-                self.hostfile_servers, self.server_group, self.basepath,
-                log_filename=self.server_log)
+            self.server_manager = ServerManager(
+                self.basepath, os.path.join(self.ompi_prefix, "bin"))
+            self.server_manager.hosts = (
+                hosts, self.workdir, self.hostfile_servers_slots)
```

Comment:
`hosts` doesn't have a value here and ends up causing this error:

```
16:56:38 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado/core/test.py:576
16:56:38 ERROR| Traceback (most recent call last):
16:56:38 ERROR|   File "/var/lib/jenkins/jenkins-1/docker_1/workspace/daos-stack_daos_PR-1008/src/tests/ftest/util/daos_core_base.py", line 53, in setUp
16:56:38 ERROR|     super(DaosCoreBase, self).setUp()
16:56:38 ERROR|   File "/var/lib/jenkins/jenkins-1/docker_1/workspace/daos-stack_daos_PR-1008/src/tests/ftest/util/apricot/apricot/test.py", line 249, in setUp
16:56:38 ERROR|     self.start_servers()
16:56:38 ERROR|   File "/var/lib/jenkins/jenkins-1/docker_1/workspace/daos-stack_daos_PR-1008/src/tests/ftest/util/apricot/apricot/test.py", line 402, in start_servers
16:56:38 ERROR|     hosts, self.workdir, self.hostfile_servers_slots)
16:56:38 ERROR| UnboundLocalError: local variable 'hosts' referenced before assignment
```

`hosts` only has a value in the other branch of this `if`/`else` because of:
https://github.com/daos-stack/daos/blob/7239f2c2e3f726be840dafc92079e70ea8b6a575/src/tests/ftest/util/apricot/apricot/test.py#L390